js-core-3/week-1/debugging-code/main.js

```diff
@@ -0,0 +1,159 @@
+const addButtons = document.querySelectorAll(".add-button");
+const formContainer = document.getElementById("form-container");
+const tableBody = document.getElementById("table-body");
+const submitted = document.getElementById("submitted");
+
+let bookNumber = 0;
+let myLibrary = [];
+
+const book1 = new Book(
+  "Crime and punishment",
+  "Fyodor Dostoyevksy",
+  671,
+  "Yes"
+);
+const book2 = new Book("A brief history of time", "Stephen Hawking", 212, "No");
+
+myLibrary.push(book1);
+myLibrary.push(book2);
+
+render();
+
+addButtons.forEach((button => {
+  button.addEventListener("click", () => {
+    formContainer.style.display = "block";
+  });
+});
+
+function addDeleteButtons() {
+  let deleteButtons = document.querySelectorAll(".delete");
+
+  deleteButtons.forEach((button) => {
+    if (button.getAttribute("data-book") == bookNumber) {
+      //Only add eventListeners to new books
+      button.addEventListener("click", () => {
+        deleteBook(button.getAttribute("data-book"));
+      });
+    }
+  });
+}
+
+function addReadButtons() {
+  let readButtons = document.querySelectorAll(".change-read");
+
+  readButtons.forEach((button) => {
+    if (button.getAttribute("data-book") == bookNumber) {
+      button.addEventListener("click", () => {
+        changeReadStatus(button.getAttribute("data-book"), button);
+      });
+    }
+  });
+}
+
+function deleteBook(number) {
+  let toDelete = document.querySelector(`tr[data-book="${number}"]`);
+  toDelete.remove();
+}
+
+function changeReadStatus(number, button) {
+  if (myLibrary[number]["read"] === "Yes") {
+    myLibrary[number]["read"] = "No";
+    button.innerText = "No";
+    button.classList.remove("button-green");
+    button.classList.add("button-red");
+  } else {
+    myLibrary[number]["read"] = "Yes";
+    button.innerText = "Yes";
+    button.classList.remove("button-red");
+    button.classList.add("button-green");
+  }
+}
+
+function Book(title, author, pages, read) {
```

Comment:
This feels like it's a pretty foreign concept to be introducing in a class where we're trying to teach them debugging techniques using only JS they should already understand?